movies-hub/src/pages/Login.jsx

```diff
@@ -0,0 +1,122 @@
+import React, { useState } from "react";
+import { Eye, EyeOff } from "lucide-react";
+import { successToast, warningToast, errorToast } from "../utils/DisplayToast";
+
+const Login = () => {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [showPassword, setShowPassword] = useState(false);
+  const [error, setError] = useState("");
+  const apiLoginUrl = import.meta.env.VITE_API_LOGIN_URL;
+  const apiKey = import.meta.env.VITE_API_KEY;
+  const apiToken = import.meta.env.VITE_API_ACCESS_TOKEN;
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    if (!email || !password) {
+      setError("Please enter both email and password.");
+      return;
+    }
+
+    try {
+      setError("");
+      const response = await fetch(apiLoginUrl, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ email, password }),
+      });
+      if (response.ok) {
+        const data = await response.json();
+        successToast("Login successful!");
+        setEmail("");
+        setPassword("");
+        console.log("Login successful:", data);
+        localStorage.setItem('accessToken', apiToken);
+      } else {
+        const errorData = await response.json();
+        console.error(
+          "Login failed. Please check your credentials.",
+          errorData.message
+        );
+        return;
+      }
+    } catch (error) {
+      setError("An error occurred. Please try again.");
+      console.error("Network error:", error);
+      return;
+    }
+  };
+
+  const handleForgetPassword = () => {
+    alert("Forgot password functionality not implemented yet.");
+  };
+
+  return (
+    <div className="min-h-screen flex items-center justify-center bg-neutral-700 px-4">
+      <form
+        onSubmit={handleSubmit}
+        className="bg-neutral-800 p-8 rounded-lg shadow-lg flex flex-col w-full max-w-sm"
+      >
+        <h2 className="text-2xl font-bold mb-4 text-white">Login</h2>
+
+        {error && <div className="text-red-600 mb-2 text-base">{error}</div>}
+
+        <input
+          id="email"
+          type="email"
+          name="email"
+          placeholder="Email"
+          value={email}
+          onChange={(e) => {
+            setEmail(e.target.value);
+            setError("");
+          }}
```

Comment:
onChnage={handleEmailChnage}

const handleEmailChnage = (e) => {    
setEmail(e.target.value);
            setError("");
}


///////////////////

onChange = {(e) => handleFieldChange(e, 'EMAIL')} 

u can use switch in this case